sync.py

```diff
@@ -32,348 +26,187 @@
+            blocks = []
+            for block in block_xml.findall('.//system-block'):
+                if any([path in block.find('path').text for path in paths_to_ignore]):
                     continue
-                # block = Block(self.cascade, block_id)
-                # read_asset returns a string version of a python dict. todo, move this to connector
-                result = self.process_block(block_id)
+
+                result = self.process_block(block.get('id'))
                 blocks.append(result)
-                yield result + newline
+                yield result + "<br/>"
                 time.sleep(time_to_wait)
-            yield newline + "All blocks have been synced."
+            yield "<br/>All blocks have been synced."
+
+            # todo: don't publish for testing. When this goes live, add this line back in
+            # self.cascade.publish(PUBLISHSET_ID, 'publishset')
+
             if send_email_after:
-                # compare hashes to SQL
-                self.check_hashes()
+                missing_data_codes = self.missing_data_codes
+
+                caps_gs_sem_email = render_template("caps_gs_sem_recipients_email.html", **locals())
+                if len(missing_data_codes) > 0:
+                    send_message("No CAPS/GS Banner Data Found", "<br/>".join(caps_gs_sem_email), html=True, caps_gs=True)
 
-                caps_gs = [MISSING_DATA_MESSAGE + "<br/>"]
-                if len(self.missing):
-                    for code in self.missing:
-                        # will there be a 2- for some reason outside of a code?
-                        if '2-' in code:
-                            caps_gs.append(code)
+                unused_banner_codes = self.get_unused_banner_codes()
+                admin_email = render_template("admin_email.html", **locals())
+                send_message("Readers Digest: Program Sync", admin_email, html=True)
 
-                caps_gs.append("<br/>If you have any questions, please email web-services@bethel.edu.")
-                if len(caps_gs) > 2:  # Only send an email to the CAPS/GS contacts if there's an errant block
-                    send_message("No CAPS/GS Banner Data Found", "<br/>".join(caps_gs), html=True, caps_gs=True)
-                self.cascade.publish(PUBLISHSET_ID, 'publishset')
-                self.create_readers_digest()
+                # reset the codes found
+                self.codes_found_in_cascade = []
 
-        return Response(stream_with_context(generator()))  # , mimetype='text/json')
+        # load the data from banner for this code
+        self.data = json.loads(requests.get('https://wsapi.bethel.edu/program-data').content)
+        return Response(stream_with_context(generator(time_to_wait, send_email_after)), mimetype='text/html')
 
     def process_block_by_path(self, path):
+        # load the data from banner for this code
+        self.data = json.loads(requests.get('https://wsapi.bethel.edu/program-data').content)
```

Comment:
Since this method calls `self.process_block_by_id`, I think setting the value of `self.data` here is unnecessary; it's the first line of process_block_by_id.

Reply:
Ah yeah thats right. I haven't gone back to refactoring the self.data out. 👍

Reply:
done.

Although you were specifically asking about that method portion, I also moved self.data out of being set as a class variable.